modules/outbound-command-event-handler/src/domain/bulk_transaction_agg.ts

```diff
@@ -0,0 +1,154 @@
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the "License") and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list (alphabetical ordering) of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+ * Modusbox
+ - Vijay Kumar Guthi <vijaya.guthi@modusbox.com>
+ --------------
+ ******/
+
+'use strict';
+
+import { ILogger } from '@mojaloop/logging-bc-public-types-lib';
+import { BaseAggregate, IEntityStateRepository } from '@mojaloop/sdk-scheme-adapter-private-shared-lib';
+import { BulkTransactionEntity, BulkTransactionState } from './bulk_transaction_entity';
+import {
+    IndividualTransferEntity,
+    IndividualTransferState,
+} from './individual_transfer_entity';
+import { IBulkTransactionEntityRepo } from '../types';
+
+export class BulkTransactionAgg extends BaseAggregate<BulkTransactionEntity, BulkTransactionState> {
+    // TODO: These counts can be part of bulk transaction entity?
+    // private _partyLookupTotalCount?: number;
+    // private _partyLookupSuccessCount?: number;
+    // private _partyLookupFailedCount?: number;
+    // // NEXT_STORY
+    // private _bulkBatches: Array<BulkBatchState>; // Create bulk batch entity
+    // private _bulkQuotesTotalCount: number;
+    // private _bulkQuotesSuccessCount: number;
+    // private _bulkQuotesFailedCount: number;
+    // private _bulkTransfersTotalCount: number;
+    // private _bulkTransfersSuccessCount: number;
+    // private _bulkTransfersFailedCount: number;
+
+    // constructor(
+    //   bulkTransactionEntity: BulkTransactionEntity,
+    //   entityStateRepo: IEntityStateRepository<BulkTransactionState>,
+    //   logger: ILogger
+    // ) {
+    //     super(bulkTransactionEntity, entityStateRepo, logger);
+    // }
+
+    static async CreateFromRequest(
```

Comment:
Referring to the base-agg comment about why we should NOT pass `bulkTransactionEntity` as part of the constructor....

1. The Aggregator is tightly coupled to its state-repo (as it should be)
2. The Aggregator should know how to load its own state, and how to save its own state...and its perfectly valid to have "NO/empty state" (which is useful for duplicate checks, etc) <-- on line 69 the Factory Function is choosing when to save the state. This should be done by the Aggregate itself based on the Command it's processing.
3. Factory functions should be considered "helpers", and should not be relied on.
4. We should not be creating an Aggregate from a Request, but rather using an Aggregate to process a request